challenges/cyclical_linked_list.py

```diff
@@ -0,0 +1,21 @@
+def has_cycle(head):
+    if is_terminated_soon(head):
+        return False
+    else:
+        return is_cyclical(head, head.next.next)
+
+def is_terminated_soon(head):
+    if head.next is None:
+        return True
+    elif head.next.next is None:
+        return True
+    else:
+        return False
+
+def is_cyclical(hop, hophop):
+    if hop == hophop:
+        return True
+    elif is_terminated_soon(hophop):
+        return False
+    else:
+        return is_cyclical(hop.next, hophop.next.next)
```

Comment:
`is_cyclical` could be compressed into a one-liner, but I think from a readability standpoint the `if/elif/else` solution is easier to comprehend.